src/symbolDocumenter.ts

```diff
@@ -16,7 +16,7 @@ const blockCommentEndRegex = /^(.*?)\s*\*\/.*$/
 
 const includeLineRegex = /^include[\s]+"([^"]+)".*$/i
 const spacerRegex = /^\s*(.)\1{3,}\s*$/
-const labelDefinitionRegex = /^[\s]*((([a-zA-Z_][a-zA-Z_0-9]*)?\.)?[a-zA-Z_][a-zA-Z_0-9]*[:]{0,2}).*$/
+const labelDefinitionRegex = /^\s*((?:[A-Z_]\w+)?(?:(?:\.[A-Z_]\w+:{0,2})|(?:[A-Z_]\w+:{1,2})))/i
```

Comment:
```suggestion
const labelDefinitionRegex = /^\s*((?:[A-Z_]\w*)?(?:(?:\.[A-Z_]\w*:{0,2})|(?:[A-Z_]\w*:{1,2})))/i
```
In general this pull request looks like a great change, but I suggest changing the instances of `\w+` to `\w*`. Unless I'm mistaken, single-character label subcomponents are legal (e.g `m:` or `m.m::`), but I believe the regex you suggest requires at least two characters: one matching `[A-Z_]` and at least one additional character matching `\w+`, so it permits `mm:` but not `m:`.

In regards to breaking backwards compatibility: in general, I prefer to maintain support for deprecated and removed RGBDS features as long as possible, so long as doing so doesn't break anything. Since the original label regex interfered with macro use sites, this is a great example of the kind of situation that justifies breaking backwards compatibility. 👍

Reply:
That is a mistake from my side, thank you for catching that!

I agree with your stance on backwards compatibility, and I only removed support for the old label syntax, as I couldn't find a better, cleaner way to go about doing it. Although, as RGBDS is still under active development (pre-1.0), breaking changes from that side are somewhat inevitable. As far as I know, the community at large are pretty good at updating their RGBDS. Even the newer verisons of hardware.inc requires >0.5.0. These are the reasons that led me to justify removing support for the old label colon-less global label declaration.